test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -14002,6 +13989,56 @@ def add_mom_dyn_res(self, custom_resource, script_body=None,
         self.add_config(a)
         return res_file
 
+    def enable_cgroup_cset(self):
+        """
+        Configure and enable cgroups hook
+        """
+        # check if cgroups subsystems including cpusets are mounted
+        with open(os.path.join(os.sep, 'proc', 'mounts'), 'r') as fd:
+            mounts = fd.read()
+        if 'cgroup' in mounts and mounts.count('cpuset') >= 2:
+            pbs_conf_val = self.du.parse_pbs_config(self.hostname)
+            f1 = os.path.join(pbs_conf_val['PBS_EXEC'], 'lib',
+                              'python', 'altair', 'pbs_hooks',
+                              'pbs_cgroups.CF')
+            vpn1 = r'"vnode_per_numa_node"\s*:\s*false,'
+            vpn2 = '"vnode_per_numa_node"   : true,'
+            uht1 = r'"use_hyperthreads"\s*:\s*false,'
+            uht2 = '"use_hyperthreads"      : true,'
+            mem1 = r'"memory"\s*:\s*{'
+            mem2 = r'"memsw"\s*:\s*{'
+            ena1 = r'"enabled"\s*:\s*true,'
+            ena2 = '"enabled"            : false,'
+            # set vnode_per_numa_node = true, use_hyperthreads = true
+            with open(f1, "r") as cfg:
+                lines = cfg.readlines()
+            for i, line in enumerate(lines):
+                if re.search(vpn1, line):
+                    lines[i] = re.sub(vpn1, vpn2, line)
+                if re.search(uht1, line):
+                    lines[i] = (re.sub(uht1, uht2, line))
+                # memory subsystem not mounted, do not enable in config
+                if (re.search(mem1, line) or re.search(mem2, line)) and \
```

Comment:
Is there something in install that sets memsw depending on whether or not the subsystem is mounted?  I didn't think there was...I believe memsw is enabled to "true" by default.  If there is no install code that changes the setting of memsw, then it shouldn't be set here either.

Reply:
Yes mem and memsw are enabled to true by default.  Setting them here as false will allow us to run PTL tests on systems that don't have memory subsystem mounted.

Reply:
I will remove setting mem and memsw to false here in the next iteration since that would not be the intended cgroups hook default setting.